### PR TITLE
fix(src/utils.ts): handle parsing quoted values with equal signs

### LINKED ISSUE #17: Can't process quoted key values with equal signs in them.
## Issue

I'm a gentledev I:
- [x] fully read the README recently
- [x] searched for existing issues
- [x] checked I'm up to date with the latest version of the project

### Expected behavior

The utility can handle keyed values containing equal signs.

### Actual behavior

The utility fails to handle keyed values with equal signs.

### Steps to reproduce the behavior

Submit a keyed value with an equal sign, such as those containing base64 encoded data. `cnonce="===="` for example.

### Debugging informations
- `node -v` result:
```
v16.16.0
```

- `npm -v` result:
```
8.16.0
```

This line here:
https://github.com/nfroidure/http-auth-utils/blob/17186d3eefb86535916d044c7f59a340bc765603/src/utils.ts#L30

Returns more than two entries for a keyed value like the following:

```
cnonce="dGVzdCBzdHJpbmc="
```

And because of the next line, https://github.com/nfroidure/http-auth-utils/blob/17186d3eefb86535916d044c7f59a340bc765603/src/utils.ts#L32, the utility throws an error.

Cnonces like this are created by Nautilus, the GNOME file manager, upon trying to connect to a WebDAV server. Using Nautilus to connect to a WebDAV or HTTP server, and using this library with the server will reliably reproduce this issue.

### PR DESCRIPTION
**Pull request by hperrin**
Fixes #17 

### Proposed changes
- Split the extracted key/value string to only two parts, returning the remainder in the last part, using splitn package.

### Code quality
- [x] I made some tests for my changes
- [x] I added my name in the
 [contributors](https://docs.npmjs.com/files/package.json#people-fields-author-contributors)
 field of the `package.json` file. Beware to use the same format than for the author field
 for the entries so that you'll get a mention in the `README.md` with a link to your website.

### License
To get your contribution merged, you must check the following.

- [x] I read the project license in the LICENSE file
- [x] I agree with publishing under this project license

### Join
- [ ] I wish to join the core team
- [x] I agree that with great powers comes responsibilities
- [x] I'm a nice person

My NPM username:

### CODE REVIEW

**Comment by hperrin** [comment]
Sorry, but NPM automatically upgraded `package-lock.json` to version 2. I didn't really change thousands and thousands of lines!

I also had to use `npm i --legacy-peer-deps` to get around some dependency issue. `metapak-nfroidure` needs to be updated to version 12 to no longer require this.

File: src/utils.ts [hunk 1]
```diff
@@ -26,19 +27,7 @@ export function parseHTTPHeadersQuotedKeyValueSet(
   if (!matches) throw new YError('E_MALFORMED_QUOTEDKEYVALUE', contents);
 
   const data = matches
-    .map((part, partPosition) => {
```
Comment (nfroidure) [inline src/utils.ts:29]: I think importing a library to just do this is overkill. Something like `return [pair[0], pair.slice(1).join('')]` will fit the needs.

**Review by nfroidure — CHANGES_REQUESTED** [review]
Otherwise LGTM

**Comment by nfroidure** [comment]
Closing in favor of https://github.com/nfroidure/http-auth-utils/pull/23

**Pull request closed without merging**